src/Neo/SmartContract/Native/NativeContract.cs

```diff
@@ -288,13 +283,26 @@ internal bool IsInitializeBlock(ProtocolSettings settings, uint index, out Hardf
 
                 if (activeIn == index)
                 {
-                    hardfork = hf;
-                    return true;
+                    hfs.Add(hf);
                 }
             }
 
+            // Return all initialize hardforks
+            if (hfs.Count > 0)
+            {
+                hardforks = hfs.ToArray();
+                return true;
+            }
+
+            // If is not configured, the Genesis is the a initialized block
+            if (index == 0 && ActiveIn is null)
+            {
+                hardforks = hfs.ToArray();
```

Comment:
I'd suggest the following since `hfs` are empty here anyway:
```suggestion
                hardforks = null;
```

Reply:
See https://github.com/neo-project/neo/pull/3202#discussion_r1580174417 for proper way

Reply:
It's not about the proper way, it's mostly for better caller's code readability since there we have:
```
if (hfs?.Length > 0)
```

Reply:
But I don't understand @AnnaShaleva, this is how it was previously, all the hardforks won't be called

Reply:
We may just keep this code as is, the behaviour itself is correct.